test/auth_scheme_with_extensions.test.ts

```diff
@@ -124,3 +113,49 @@ test.each(tokenVectors)('AuthScheme-TokenVector-%#', async (v: TokenVectors) =>
 
     expect(uint8ToHex(authInputEnc)).toBe(v.token_authenticator_input);
 });
+
+describe('extensions', () => {
+    test('maxSizeExtension', () => {
+        const ext = new Extension(0xaa, new Uint8Array(MAX_UINT16));
+        testSerializeWithOps(Extension, ext);
+
+        const extBytes = ext.serialize();
+        expect(extBytes.length).toBe(2 + 2 + MAX_UINT16);
+    });
+
+    test('maxSizeArrayExtension', () => {
```

Comment:
this test does not pass

Reply:
this is intended so it shows the issue. feel free to cherry pick and fix with the solution agreed by the spec authors